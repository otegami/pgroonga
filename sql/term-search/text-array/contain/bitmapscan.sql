CREATE TABLE tags (
  names text[]
);

INSERT INTO tags VALUES (ARRAY['PostgreSQL', 'PG']);
INSERT INTO tags VALUES (ARRAY['Groonga', 'grn', 'groonga']);
INSERT INTO tags VALUES (ARRAY['PGroonga', 'pgrn', 'groonga']);

CREATE INDEX pgroonga_index ON tags
  USING pgroonga (names pgroonga_text_array_term_search_ops_v2);

SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;

EXPLAIN (COSTS OFF)
SELECT names
  FROM tags
 WHERE names @> ARRAY['grn', 'groonga'];

SELECT names
  FROM tags
 WHERE names @> ARRAY['grn', 'groonga'];

DROP TABLE tags;
