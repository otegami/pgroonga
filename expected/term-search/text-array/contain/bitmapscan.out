CREATE TABLE tags (
  names text[]
);
INSERT INTO tags VALUES (ARRAY['PostgreSQL', 'PG']);
INSERT INTO tags VALUES (ARRAY['Groonga', 'grn', 'groonga']);
INSERT INTO tags VALUES (ARRAY['PGroonga', 'pgrn', 'groonga']);
CREATE INDEX pgroonga_index ON tags
  USING pgroonga (names pgroonga_text_array_term_search_ops_v2);
SET enable_seqscan = off;
SET enable_indexscan = off;
SET enable_bitmapscan = on;
EXPLAIN (COSTS OFF)
SELECT names
  FROM tags
 WHERE names @> ARRAY['grn', 'groonga'];
                       QUERY PLAN                       
--------------------------------------------------------
 Bitmap Heap Scan on tags
   Recheck Cond: (names @> '{grn,groonga}'::text[])
   ->  Bitmap Index Scan on pgroonga_index
         Index Cond: (names @> '{grn,groonga}'::text[])
(4 rows)

SELECT names
  FROM tags
 WHERE names @> ARRAY['grn', 'groonga'];
         names         
-----------------------
 {Groonga,grn,groonga}
(1 row)

DROP TABLE tags;
